src/shared/pam-util.c

```diff
@@ -50,6 +50,18 @@ int pam_syslog_pam_error(pam_handle_t *handle, int level, int error, const char
 }
 
 static void cleanup_system_bus(pam_handle_t *handle, void *data, int error_status) {
+        /* The PAM_DATA_SILENT flag is the way that pam_end() communicates to the module stack that this
+         * invocation of pam_end() is not the final one, but in the process that is going to directly exec
+         * the child. This means we are being called after a fork(), and we do not want to try and clean
+         * up the sd-bus object, as the file descriptors (which are CLOEXEC) will be invalid, and we'll hit
+         * an assertion. */
+        if (error_status & PAM_DATA_SILENT)
+                return (void) pam_syslog_pam_error(
+                                handle,
+                                LOG_ERR,
+                                SYNTHETIC_ERRNO(EUCLEAN),
+                                "Attempted to close sd-bus after fork, this should not happen.");
+
```

Comment:
so when the PAM program that allocates the session sets a flag called "silent" we become chatty? that's certainly surprising at least...

Reply:
It's a programming error somewhere, so it should be an assert ideally, but given how delicate the area is, I figure it's better to start from complaining loudly first. Shake the tree and see what falls off.